sdk/openai/openai/sources/generated/src/api/models.ts

```diff
@@ -164,48 +166,48 @@ export interface ChatChoice {
   delta?: ChatMessage;
 }
 
-/** */
-export interface Embeddings {
-  /** Embedding values for the prompts submitted in the request. */
-  data: EmbeddingItem[];
-  /** Usage counts for tokens input using the embeddings API. */
-  usage: EmbeddingsUsage;
-}
-
-/** */
-export interface Completions {
-  /** A unique identifier associated with this completions response. */
+/** A polling status update or final response payload for an image operation. */
+export interface BatchImageGenerationOperationResponse {
+  /** The ID of the operation. */
   id: string;
-  /**
-   * The first timestamp associated with generation activity for this completions response,
-   * represented as seconds since the beginning of the Unix epoch of 00:00 on 1 Jan 1970.
-   */
+  /** A timestamp when this job or item was created (in unix epochs). */
   created: number;
-  /**
-   * The collection of completions choices associated with this completions response.
-   * Generally, `n` choices are generated per provided prompt with a default value of 1.
-   * Token limits and other settings may limit the number of choices generated.
-   */
-  choices: Choice[];
-  /** Usage information for tokens processed and generated as part of this completions operation. */
-  usage: CompletionsUsage;
+  /** A timestamp when this operation and its associated images expire and will be deleted (in unix epochs). */
+  expires?: number;
+  /** The result of the operation if the operation succeeded. */
+  result?: ImageGenerations;
+  /** The status of the operation */
+  status: AzureOpenAIOperationState;
+  /** The error if the operation failed. */
+  error?: ErrorModel;
 }
 
-/** */
-export interface ChatCompletions {
-  /** A unique identifier associated with this chat completions response. */
-  id: string;
-  /**
-   * The first timestamp associated with generation activity for this completions response,
-   * represented as seconds since the beginning of the Unix epoch of 00:00 on 1 Jan 1970.
-   */
+/** The result of the operation if the operation succeeded. */
+export interface ImageGenerations {
+  /** A timestamp when this job or item was created (in unix epochs). */
   created: number;
-  /**
-   * The collection of completions choices associated with this completions response.
-   * Generally, `n` choices are generated per provided prompt with a default value of 1.
-   * Token limits and other settings may limit the number of choices generated.
-   */
-  choices: ChatChoice[];
-  /** Usage information for tokens processed and generated as part of this completions operation. */
-  usage: CompletionsUsage;
+  /** The images generated by the operator. */
+  data: ImageLocation[] | ImagePayload[];
```

Comment:
Interesting union here. Our API should probably add custom API methods that allow the developer to know which type (ImageLocation[] or ImagePayload[]) they will be receiving in the response.

Reply:
Yeah, I was thinking it would be better to have two convenience methods (one for each ImageGenerationResponseFormat) that return the specific type requested instead of trying to infer the result type from the input. Does that make sense to you?